spring-cloud-dataflow-module-deployers/spring-cloud-dataflow-module-deployer-cloudfoundry/src/main/java/org/springframework/cloud/dataflow/module/deployer/cloudfoundry/ModuleStatusBuilder.java

```diff
@@ -56,9 +56,15 @@ ModuleStatusBuilder withId(ModuleDeploymentId id) {
 		return this;
 	}
 
-	// TODO: Find out what the application state candidates are
 	private static ModuleStatus.State convertState(String state) {
 		switch (state) {
+			case "STARTED":
+			case "STARTING":
+			case "STOPPED":
```

Comment:
Thanks Paul. Based on that description, I would say "deploying" in indeed the right mapping.

At some point we might want to consider the option of a "stream undeploy" triggering apps to STOP instead of always being a DELETE. We would probably need to bubble up the distinction in some way (STOP would be preferable when you will later want to re-deploy with minimal overhead, but DELETE is required I assume if you want to change any deployment properties/env vars when re-deploying). AFAIK, we don't have the same distinction on Lattice where *stream deploy* maps to CREATE and *stream undeploy* maps to REMOVE.